File: java/src/test/java/ai/buble/sdk/AppsTest.java
package ai.buble.sdk;

import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

final class AppsTest {
    @Test
    void createsAppGenerationWithEncodedPath() throws Exception {
        try (TestHttpServer server = new TestHttpServer(request ->
                new TestHttpServer.Response(200, "{\"data\":{\"id\":\"app_task_1\",\"status\":\"pending\"}}"))) {
            BubleClient client = BubleClient.builder().apiKey("sk_test").baseUrl(server.url()).build();

            client.apps().generations().create("video/background remover", Map.of(
                    "source_video", List.of("https://example.com/source.mp4"),
                    "refine_foreground_edges", true
            ));

            assertEquals("/api/v1/apps/video%2Fbackground%20remover/generations", server.requests().get(0).path);
            assertTrue(server.requests().get(0).body.contains("\"source_video\""));
        }
    }
}

File: ruby/test/apps_test.rb
# frozen_string_literal: true

require_relative 'test_helper'

class AppsTest < Minitest::Test
  def test_retrieves_app_with_encoded_path
    transport = FakeTransport.new
    transport.enqueue_response('data' => {})
    resource = Buble::AppsResource.new(transport)

    resource.retrieve('video background')

    assert_equal '/api/v1/apps/video%20background', transport.requests[0][:path]
  end

  def test_creates_app_generation_with_flat_params
    transport = FakeTransport.new
    transport.enqueue_response('data' => { 'id' => 'task_1' })
    resource = Buble::AppsResource.new(transport)

    resource.generations.create('video-background-remover', {
                                  'source_video' => ['https://example.com/source.mp4'],
                                  'subject_is_person' => true
                                })

    assert_equal '/api/v1/apps/video-background-remover/generations', transport.requests[0][:path]
    assert_equal true, transport.requests[0][:body]['subject_is_person']
  end
end
